domain-models-runtime/src/test/java/org/folio/rest/persist/PostgresClientIT.java

```diff
@@ -643,6 +648,40 @@ public void updateSectionSingleQuote(TestContext context) {
     }));
   }
 
+    @Test
+  public void selectWithTimeoutSuccess(TestContext context) {
+      PostgresClient client = postgresClient();
+      client.getSQLConnection(2000, conn -> {
+        client.selectSingle(conn, "SELECT 1, pg_sleep(1);", context.asyncAssertSuccess());
+      });
+  }
+
+  @Test
+  public void selectWithTimeoutFailure(TestContext context) {
+      PostgresClient client = postgresClient();
+      client.getSQLConnection(1000, conn -> {
+        client.selectSingle(conn, "SELECT 1, pg_sleep(2);", context.asyncAssertFailure());
+      });
```

Comment:
```suggestion
      client.getSQLConnection(1000, asyncAssertTx(context, conn -> {
        client.selectSingle(conn, "SELECT 1, pg_sleep(2);", context.asyncAssertFailure(e -> {
          String sqlState = new PgExceptionFacade(e).getSqlState();
          assertThat(PgExceptionUtil.getMessage(e), sqlState, is("57014"));  // query_canceled
        }));
      }));
```
Each async callback needs a context.asyncAssertSuccess or context.asyncAssertFail or asyncAssertTx. Otherwise the test will succeed without executing the SELECT.

Please check the sqlState returned.